src/node/blockmanager_args.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/blockmanager_args.h>
+
+#include <util/system.h>
+#include <validation.h>
+
+namespace node {
+std::optional<bilingual_str> ApplyArgsManOptions(const ArgsManager& args, BlockManager::Options& opts)
+{
+    // block pruning; get the amount of disk space (in MiB) to allot for block & undo files
+    int64_t nPruneArg = args.GetIntArg("-prune", 0);
```

Comment:
I'm taking a look at this with some fresh knowledge now. If I understand this correctly, this always overrides what is passed in the `opts` with `0` if the `prune` arg is not set. The way I understand the current patterns elsewhere in the code for reading the arguments, the passed in opts are preserved if no argument is found, for example as done in [chainstatemanager_args](https://github.com/bitcoin/bitcoin/blob/master/src/node/chainstatemanager_args.cpp) / [chainstatemanager_opts](https://github.com/bitcoin/bitcoin/blob/master/src/kernel/chainstatemanager_opts.h). I'm not sure which pattern is preferable to be honest, but I think it should be consistent.

This also similarly applies to `fPruneMode`.

Reply:
Right on, can be marked as resolved.